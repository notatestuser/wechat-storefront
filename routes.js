const nextRoutes = require('next-routes');

const routes = module.exports = nextRoutes();

routes.add({ name: 'index', pattern: '/' });
routes.add({ name: 'login', pattern: '/login' });
routes.add({ name: 'logout', pattern: '/logout' });
routes.add({ name: 'products', pattern: '/products' });
