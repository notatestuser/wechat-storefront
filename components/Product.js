import React from 'react';
import striptags from 'striptags';
import styled from 'styled-components';
import { Card, Icon } from 'antd';
import { Link } from '../routes';

const { Meta } = Card;

const generateDescriptionHtml = (html) => ({ __html: striptags(html, ['h4', 'p', 'strong']) });

const StyledProduct = styled(Card)`
  display: flex;
  flex-direction: column;
  margin-bottom: 15px;
  width: 300px;
`;

const ProductTitle = styled.h4`
  margin-bottom: 12px;
`;

const CoverImage = styled.img`
  max-height: 200px;
  object-fit: cover;
  object-position: center top;
`;

export default (product) => (
  <StyledProduct
    key={product.id}
    bodyStyle={{ flex: 1 }}
    cover={<CoverImage alt="example" src={product.thumbnail} />}
    actions={[
      <Link route={`/product/${product.id}`} prefetch={true}>
        <a><Icon type="info-circle" /></a>
      </Link>,
      <Icon type="wechat" />,
      <Icon type="ellipsis" />,
    ]}
  >
    <Meta
      title={<ProductTitle>{product.title}</ProductTitle>}
      description={<div
        dangerouslySetInnerHTML={generateDescriptionHtml(product.description)}
      />}
    />
  </StyledProduct>);
