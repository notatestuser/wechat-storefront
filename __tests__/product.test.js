/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Icon } from 'antd';

import { Link } from '../routes';
import Product from '../components/Product';

describe('<Product />', () => {
  it('renders correctly', () => {
    const product = {
      description: 'Product Description',
      id: 'abc',
      thumbnail: 'https://lukep.org/placeholder.png',
      title: 'Product Title',
    };
    const component = renderer.create(<Product {...product} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('links to the product page', () => {
    expect.assertions(2);
    const product = {
      description: 'Product Description',
      id: 'abc',
      thumbnail: 'https://lukep.org/placeholder.png',
      title: 'Product Title',
    };
    const component = mount(<Product {...product} />);
    component.find(Link).forEach(node => {
      const actualProps = node.props();
      expect(actualProps).toEqual({
        route: '/product/abc',
        prefetch: true,
        children: actualProps.children,
      });
    });
  });

  it('contains a wechat icon', () => {
    const component = mount(<Product />);
    const actual = component.contains(<Icon type="wechat" />);
    expect(actual).toEqual(true);
  });
});
