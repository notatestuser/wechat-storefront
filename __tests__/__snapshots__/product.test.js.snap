// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> renders correctly 1`] = `
<div
  className="ant-card sc-bdVaJa crvomV ant-card-bordered"
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="example"
      className="sc-htpNat hcQtxM"
      src="https://lukep.org/placeholder.png"
    />
  </div>
  <div
    className="ant-card-body"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <div>
      <div
        className="ant-card-meta"
      >
        <div
          className="ant-card-meta-detail"
        >
          <div
            className="ant-card-meta-title"
          >
            <h4
              className="sc-bwzfXH DZENQ"
            >
              Product Title
            </h4>
          </div>
          <div
            className="ant-card-meta-description"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Product Description",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <ul
    className="ant-card-actions"
  >
    <li
      style={
        Object {
          "width": "33.333333333333336%",
        }
      }
    >
      <span>
        <a
          href="/product/abc"
          onClick={[Function]}
        >
          <i
            className="anticon anticon-info-circle"
          />
        </a>
      </span>
    </li>
    <li
      style={
        Object {
          "width": "33.333333333333336%",
        }
      }
    >
      <span>
        <i
          className="anticon anticon-wechat"
        />
      </span>
    </li>
    <li
      style={
        Object {
          "width": "33.333333333333336%",
        }
      }
    >
      <span>
        <i
          className="anticon anticon-ellipsis"
        />
      </span>
    </li>
  </ul>
</div>
`;
